### PR TITLE
Vi/2fa guard

### LINKED ISSUE #66: USER ACCOUNT - 2FA
- [ ] User table needs to be altered to save:
  - boolean for 2FA on/off
  - string for 2FA secret - encrypted
  - [ ] Figure out how to save the secret in the database (encrypted?)
- [ ] User needs an option to turn on/off 2FA in their account (needs frontend/backend task)
- [ ] Login flow needs to be altered to include a check to 2FA boolean + a view to input the 6 digit token

How 2FA would work:
- User turns on 2FA in profile
- Users gets send to a setup view where they are shown a QR code to scan with their OTP app
- Once they have scanned/saved the code they can click "next" and are asked to input a 6 digit code to verify
- If the code is correct we save that 2FA is on and their secret in our DB

Some package options
- https://www.npmjs.com/package/2fa-util (definitely generates QR codes!)
- ~~https://www.npmjs.com/package/@multiotp/genotp~~ (doesn't gen. QR codes)
- ~~https://www.npmjs.com/package/totp-generator~~ (doesn't gen. QR codes)

### PR DESCRIPTION
**Pull request by vincentvis**
2fa done

changed the auth guard to check more

added 2fa to login flow

closes #66

### CODE REVIEW

**Review by Ryno95 — APPROVED** [review]
Works super smoothly!

I only have some curiosity comments/questions :)

File: backend/src/auth/auth.guard.ts [hunk 1]
```diff
@@ -1,22 +1,53 @@
...
-		} catch (e) {
-			return false;
+		async canActivate(context: ExecutionContext): Promise<boolean> {
+			const request = context.switchToHttp().getRequest<Request>();
+			try {
+				const jwt = request.cookies['jwt'];
+				const validated: JwtDto = this.jwtService.verify(jwt);
+
+				const expiryTimestamp: number = validated.exp * 1000;
+				const nowTimestamp = +new Date();
```
[the remark this answers is not in the document]
Comment (vincentvis) [reply backend/src/auth/auth.guard.ts:28]: the + operator converts things to numbers
so it's kinda `+(new Date())`
where `new Date()` returns a date object, and the `+` converts it to an unix timestamp

File: backend/src/auth/auth.guard.ts [hunk 2]
```diff
@@ -1,22 +1,53 @@
...
-			return validated;
-		} catch (e) {
-			return false;
+		async canActivate(context: ExecutionContext): Promise<boolean> {
+			const request = context.switchToHttp().getRequest<Request>();
+			try {
+				const jwt = request.cookies['jwt'];
+				const validated: JwtDto = this.jwtService.verify(jwt);
+
+				const expiryTimestamp: number = validated.exp * 1000;
```
[the remark this answers is not in the document]
Comment (vincentvis) [reply backend/src/auth/auth.guard.ts:27]: yes
the jwt stores the timestamp in a less precise time than what you get from javascript's `new Date()`

File: backend/src/auth/auth.guard.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ryno95) [reply backend/src/auth/auth.guard.ts:28]: Ooooh yea remember it now from the generated CRUD endpoints haha

**Pull request merged**